README.md

```diff
@@ -31,3 +31,53 @@ For people who are trying to modify the image stack, here are some scenarios and
 4. If the workflow failed, go to the action details page and debug the issue. Push a new change by amending the first commit `git add . && git commit --amend`.
 5. If the workflow completed successfully (a green check to the left of the commit), you can now safely merge the pull request to the `main` branch.
 6. The workflow will now run again and push the images to Dockerhub.
+
+
+### Overview of the Repository
+We Use github workflow to builds new images if their is any change in the images or addtional images are added.<br>
+.github:-<br>
+        The main.yml file contains the steps to build the image<br>
+            - Initial download required for the images are installed<br>
+            - Doit module is used to run the tasks<br>
+                1) install all the requirments for the images<br>
+                2) Use doit unit build to build,test and push the image to DataHub repo<br>
+                3) Update the wiki  <br>
+                4) Store the artifacts and logs<br>
+        The Tag.yml file is used to tag the latest build <br>
+<br>
+Images:-<br>
+        1) Folder containing the images and each images has its own docker file, test folder for test scripts.
+        2) Spec.yml file is important file where the dependecy of the image and build information specified for an image
+
+Model:-
+        1) Spec.py used for reading the spec file and prepare build parameters<br>
+        2) imagedef.py pydantic object used by spec.py<br>
+scripts:-<br>
+        1) dataojects.py contains the pydantic class<br>
+        2) docker_unit.py used to build,test and push images<br>
+            - ContainerFacade class controls the flow of the code<br>
+            - build_test_push_containers method is used for building,test and push image<br>
+        3) githelper has code for git related task<br>
+        4) manifests.py has utils methods to maintain the manifests<br>
+        5) Other docker file will be deprecated except for manifest<br>
+Tests:-<br>
+        1) Contains the test file for testing the submodule<br>
+        2) used with pytest can be called with pytest test_*.py<br>
+dodo.py:-<br>
+        1) We use doit python module to call the submodule <br>
+        2) Each task is called with parameters and expected outputs<br>
+<br>
+Steps to build locally:<br>

```

Comment:
Break this into a separate header ### and move it to the beginning of ## Maintenance